### PR TITLE
Bug fixes and websocket caching

### PR DESCRIPTION
**Pull request by Ekwav**
Fixed uuid, auction path and search flicker.
Websocket responses are now cached and reused.

### CODE REVIEW

File: src/api/ApiHelper.tsx [hunk 1]
```diff
@@ -133,6 +151,7 @@ function initAPI(): API {
                 type: RequestType.ALL_ENCHANTMENTS,
                 data: "",
                 resolve: (enchantments: any) => {
+                    console.log(enchantments);
```
Comment (matthias-luger) [inline src/api/ApiHelper.tsx:154]: remove console.log

File: src/api/WebsocketHelper.tsx [hunk 2]
```diff
@@ -55,15 +61,29 @@ function initWebsocket(): WebSocket {
     return getNewWebsocket();
 }
 
-function sendRequest(request: ApiRequest): void {
+async function sendRequest(request: ApiRequest): Promise<void> {
```
Comment (matthias-luger) [inline src/api/WebsocketHelper.tsx:64]: change to Promise (consistency)

File: src/api/WebsocketHelper.tsx [hunk 3]
```diff
@@ -55,15 +61,29 @@ function initWebsocket(): WebSocket {
...
+    let requestJson = JSON.stringify(request.data);
+    var cacheValue = await get(request.type + requestJson)
```
Comment (matthias-luger) [inline src/api/WebsocketHelper.tsx:66]: extract cache logic into own file (e.g. utils/CacheUtils)
get, set are unintuitive functions

File: src/components/AuctionList/AuctionList.tsx [hunk 4]
```diff
@@ -27,23 +27,24 @@ function AuctionList(props: Props) {
 
     let loadNewAuctions = (): void => {
         api.getAuctions(props.playerUUID, 20, auctions.length).then(newAuctions => {
+            console.log(newAuctions)
```
Comment (matthias-luger) [inline src/components/AuctionList/AuctionList.tsx:30]: remove console.log

File: src/api/WebsocketHelper.tsx [hunk 5]
```diff
@@ -34,6 +36,10 @@ function initWebsocket(): WebSocket {
             request.reject(response.data);
         } else {
             request.resolve(JSON.parse(response.data));
+            // cache the response 
+            if (!request.type.includes("subscribe")) {
```
Comment (matthias-luger) [inline src/api/WebsocketHelper.tsx:40]: extract logic in seperate file (e.g. utils/CacheUtils)

File: src/api/WebsocketHelper.tsx (hunk 3 above)
Comment (matthias-luger) [inline src/api/WebsocketHelper.tsx:66]: use let

File: src/components/AuctionList/AuctionList.tsx [hunk 6]
```diff
@@ -58,13 +59,19 @@ function AuctionList(props: Props) {
 
     let getItemImageElement = (auction: Auction) => {
         return (
-            auction.item.iconUrl ? <img className="auction-item-image" src={auction.item.iconUrl} alt="" height="48" width="48" /> : undefined
+            auction.item.iconUrl ? <img className="auction-item-image" src={auction.item.iconUrl} alt="" height="48" width="48" onError={(error)=>onImageLoadError(auction,error)}/> : undefined
         )
     }
 
+    let onImageLoadError = (auction: Auction,data:any) => {
```
Comment (matthias-luger) [inline src/components/AuctionList/AuctionList.tsx:66]: better logs

File: src/components/Search/Search.tsx [hunk 7]
```diff
@@ -41,14 +38,15 @@ function Search(props: Props) {
         if (searchDebounce) {
             clearTimeout(searchDebounce);
         }
-        if (newSearchText === "") {
+        // there is a search response for "", it contains the most popular overall
```
Comment (matthias-luger) [inline src/components/Search/Search.tsx:41]: no search for ""

File: src/utils/Parser/APIResponseParser.tsx [hunk 8]
```diff
@@ -3,7 +3,8 @@ export function parseItemBid(bid: any): ItemBid {
         uuid: bid.uuid,
         end: new Date(bid.end),
         item: {
-            name: bid.itemName
+            name: bid.itemName,
```
Comment (matthias-luger) [inline src/utils/Parser/APIResponseParser.tsx:6]: name doesnt exist anymore

File: .vscode/launch.json [hunk 9]
```diff
@@ -4,6 +4,7 @@
     // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (matthias-luger) [inline .vscode/launch.json:4]: add launch.json to gitignore
and remove current change

File: src/components/Search/Search.tsx (hunk 7 above)
Comment (matthias-luger) [reply src/components/Search/Search.tsx:41]: remove comment

File: src/utils/Parser/APIResponseParser.tsx [hunk 10]
```diff
@@ -3,7 +3,8 @@ export function parseItemBid(bid: any): ItemBid {
         uuid: bid.uuid,
         end: new Date(bid.end),
         item: {
-            name: bid.itemName
```
Comment (matthias-luger) [inline src/utils/Parser/APIResponseParser.tsx:6]: set name and tag everywhere

File: .vscode/launch.json [hunk 11]
```diff
@@ -1,15 +0,0 @@
-{
```
Comment (matthias-luger) [inline .vscode/launch.json:1]: pls revert file to the state in master

File: src/components/PriceGraph/PriceGraph.tsx [hunk 12]
```diff
@@ -44,7 +44,7 @@ function PriceGraph(props: Props) {
                 return (a as number) - (b as number);
             });
             priceChart!.options.title!.text =
-                "Price for 1 " + props.item.name;
+                "Price for 1 " + props.item.tag;
```
Comment (matthias-luger) [inline src/components/PriceGraph/PriceGraph.tsx:47]: dont show the tag to the user
show the itemName

File: src/pages/ItemDetails/ItemDetails.tsx [hunk 13]
```diff
@@ -12,7 +12,7 @@ function ItemDetails() {
 
     let getItem = (): Item => {
         return {
-            name: itemName
+            tag: itemName
```
Comment (matthias-luger) [inline src/pages/ItemDetails/ItemDetails.tsx:15]: change Variable and routing name to itemTag (not itemName)

File: src/routes.tsx [hunk 14]
```diff
@@ -17,6 +17,6 @@ export default (
     <Route exact path="/" component={() => <Redirect to="/item/Aspect of the End" />}/>
     <Route path='/player/:uuid' component={PlayerDetails}/>
     <Route path='/item/:itemName' component={ItemDetails}/>
```
Comment (matthias-luger) [inline src/routes.tsx:19]: change Routing-Variable to itemTag

File: src/utils/Parser/APIResponseParser.tsx [hunk 15]
```diff
@@ -56,7 +59,7 @@ export function parseItemPriceData(priceData: any): ItemPriceData {
 
 export function parseItem(item: any): Item {
     return {
-        name: item.Name,
+        tag: item.Name,
```
Comment (matthias-luger) [inline src/utils/Parser/APIResponseParser.tsx:62]: dont set the tag to item.Name

File: src/utils/Parser/APIResponseParser.tsx [hunk 16]
```diff
@@ -40,7 +43,7 @@ export function parsePlayerDetails(playerDetails: any): PlayerDetails {
                 highestBid: auction.highestBid,
                 end: new Date(auction.end),
                 item: {
-                    name: auction.itemName
+                    tag: auction.itemName
```
Comment (matthias-luger) [inline src/utils/Parser/APIResponseParser.tsx:46]: dont set the tag to item.Name

File: src/utils/Parser/APIResponseParser.tsx (hunk 10 above)
Comment (Ekwav) [reply src/utils/Parser/APIResponseParser.tsx:6]: exists

**Pull request merged**